src/PubSubClient.cpp

```diff
@@ -584,15 +581,15 @@ void PubSubClient::disconnect() {
 }
 
 uint16_t PubSubClient::writeString(const char* string, uint8_t* buf, uint16_t pos) {
-    const char* idp = string;
-    uint16_t i = 0;
-    pos += 2;
-    while (*idp) {
-        buf[pos++] = *idp++;
-        i++;
+    uint16_t copiedSymbolsNum = 0;
+    uint16_t lengthPosition = pos;
+    pos += 2;   //hold place for length (2 bytes)
+    while (*string == '\0') {
+        buf[pos++] = *string++;
```

Comment:
Just a general remark.
Both in your change as in the original code is no check for the size of the array `buf`.